packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -217,6 +217,7 @@ class _DayPickerGridDelegate extends SliverGridDelegate {
       crossAxisStride: tileWidth,
       childMainAxisExtent: tileHeight,
       childCrossAxisExtent: tileWidth,
+      crossAxisExtent: constraints.crossAxisExtent,
```

Comment:
You'll have to add a separate test for this fix to ensure that we never again break this in the date picker.

(Also, it would be good to include some context about this specific fix in the PR description. I was pretty surprised to find a change to date picker in this PR).

Reply:
I've updated the PR description to explain this extra parameter.

The date picker implements the SliverGridDelegate interface https://github.com/flutter/flutter/blob/64f02fe168955b21d5abe6172166f05754911849/packages/flutter/lib/src/material/date_picker.dart#L206 and uses a GridView to display it. https://github.com/flutter/flutter/blob/64f02fe168955b21d5abe6172166f05754911849/packages/flutter/lib/src/material/date_picker.dart#L495

So either the GridView test covers the use of GridView in the framework, or perhaps the test should be on the SliverGridLayout implementation use.

Reply:
There needs to be a test that fails when this line is deleted.

There are tests in the framework to ensure that SliverGridRegularTileLayout works correctly. But this needs to be a tests that ensures that _DayPickerGridDelegate uses it correctly.